pandas/core/indexing.py

```diff
@@ -2165,6 +2164,18 @@ def _setitem_single_column(self, loc: int, value, plane_indexer) -> None:
         else:
             # set value into the column (first attempting to operate inplace, then
             #  falling back to casting if necessary)
+            dtype = self.obj.dtypes.iloc[loc]
+            if dtype == np.void:
+                # This means we're expanding, with multiple columns, e.g.
+                #     df = pd.DataFrame({'A': [1,2,3], 'B': [4,5,6]})
+                #     df.loc[df.index <= 2, ['F', 'G']] = (1, 'abc')
+                # Columns F and G will initially be set to np.void.
+                # Here, we replace those temporary `np.void` columns with
+                # columns of the appropriate dtype, based on `value`.
+                arr = sanitize_array(value, Index(range(1)), copy=False)
```

Comment:
`construct_1d_arraylike_from_scalar`?

Reply:
Discussed during the dev meeting. If the idea is to fill with the appropriate missing value, I think you can use a combination of `infer_fill_value` with `construct_1d_arraylike_from_scalar`

Reply:
thanks! will update

Reply:
I think this would run into the same issues that https://github.com/pandas-dev/pandas/pull/56321 addressed

I'd factored out the common logic, if there's a better way perhaps it can be addressed separately?